cmd/admxgen/defs/apparmor.yaml

```diff
@@ -16,3 +16,37 @@
   type: "apparmor"
   meta:
     strategy: append
+- key: "/apparmor-users"
+  displayname: "AppArmor"
+  explaintext: |
+    Define an AppArmor user profile to be parsed and loaded on client machines.
+    The profile is specified as a file path relative to the SYSVOL/apparmor/ubuntu/ directory.
+    On the client machine, user profiles are stored in /etc/apparmor.d/adsys/users/<user-name>, thus the administrator can reference abstractions and tunables shipped with the client distribution of AppArmor.
+
+    The profile will ideally contain a mapping between a user and a role. Roles must be configured beforehand in the System-wide application confinement section.
+    Below is an example of a user profile declaration:
+
+      ^{{ADSYS_USER}} {
+        include <abstractions/authentication>
+        include <abstractions/nameservice>
+
+        capability dac_override,
+        capability setgid,
+        capability setuid,
+        /etc/default/su r,
+        /etc/environment r,
+        @{HOMEDIRS}/.xauth* w,
+        /usr/bin/{,b,d,rb}ash Px -> default_user,
+        /usr/bin/{c,k,tc}sh Px -> default_user,
+      }
+
+    Where {{ADSYS_USER}} is a special string that will be replaced with the client username when adsys applies the policy, and default_user is a role declared beforehand in the Machine section. More details and examples can be found in the apparmor section of the adsys documentation.
+
+    The configured profile will override any profile referenced higher in the GPO hierarchy.
+  elementtype: "text"
+  release: "any"
+  note: |
+   -
+    * Enabled: The profile in the text entry is applied on the client machine.
+    * Disabled: The profile is removed from the target machine, and any related rules are unloaded.
```

Comment:
unconfigured vs Disabled, is there any change of behaviour? (seems that if Disabled and parent will configured something, we won’t have it compared to only unconfigured. Is it worth mentionning?)

Reply:
with inheritance, I’m really sure that if you have a "closer" GPO as unconfigured, it’s just ignored and the parent wins… Where disable will override the parent.

The unconfigured case works if all of them are unconfigured.